pycbc/inject/inject.py

```diff
@@ -455,11 +457,14 @@ class InjectionSet(object):
 
     def __init__(self, sim_file, **kwds):
         ext = os.path.basename(sim_file)
-        if ext.endswith(('.xml', '.xml.gz', '.xmlgz')):
-            self._injhandler = _XMLInjectionSet(sim_file, **kwds)
+        if ext.endswith(('.xml', '.xml.gz')):
+            self._injhandler = _XMLInjectionSet(sim_file, **kwds) 
             self.indoc = self._injhandler.indoc
         elif ext.endswith(('hdf', '.h5')):
             self._injhandler = _HDFInjectionSet(sim_file, **kwds)
+        elif ext.endswith(('.xmlgz')): ## If the injection is based on ASCII file and uses the distribution of the inspinj, please use this extention (sorry for this terrible way)
```

Comment:
Why not just add the switch based on the waveform into the xml injection set? Is there a reason we need to duplicate the class?

Reply:
Agreed, added in the same class a condition, if the approximant name is 'ascii' it takes injections from ascii file